ztools/telnet/telnet.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * ZGrab Copyright 2015 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package telnet
+
+import (
+	"bytes"
+	"errors"
+	"net"
+)
+
+var (
+	IAC                = byte(255) //Interpret as command
+	DONT               = byte(254)
+	DO                 = byte(253)
+	WONT               = byte(252)
+	WILL               = byte(251)
+	IAC_CMD_LENGTH     = 3 // IAC commands take 3 bytes (inclusive)
+	READ_BUFFER_LENGTH = 8192
+)
+
+func GetTelnetBanner(logStruct *TelnetLog, conn net.Conn) error {
+	var err error
+	if err = NegotiateOptions(logStruct, conn); err != nil {
+		return err
+	}
+
+	//grab banner
+	buffer := make([]byte, READ_BUFFER_LENGTH)
+
+	numBytes := len(buffer)
+	count := 0
+	for numBytes != 0 && count < 15 {
```

Comment:
I was artificially capping the length of the banner data response, but I will create a new flag telnet-max-size defaulting to a sane value that will determine how much of the telnet banner to read.